api/src/main/java/com/epam/pipeline/manager/billing/BillingHelper.java

```diff
@@ -219,35 +234,115 @@ public TopHitsAggregationBuilder aggregateLastByDateDoc() {
-        return getLongSum(aggregations, BillingHelper.COST_FIELD);
+    public FilterAggregationBuilder aggregateFilteredRuns() {
+        return AggregationBuilders.filter(BillingUtils.RUN_COST_AGG, queryRunTerms());
+    }
+
+    public FilterAggregationBuilder aggregateFilteredStorages() {
+        return AggregationBuilders.filter(BillingUtils.STORAGE_COST_AGG, queryStorageTerms());
+    }
+
+    public AvgAggregationBuilder aggregateStorageUsageAvg() {
+        return AggregationBuilders.avg(BillingUtils.STORAGE_USAGE_AGG)
+                .field(BillingUtils.STORAGE_USAGE_FIELD);
+    }
+
+    public BucketSortPipelineAggregationBuilder aggregateCostSortBucket() {
+        return PipelineAggregatorBuilders.bucketSort(BillingUtils.SORT_AGG, Collections.singletonList(
+                new FieldSortBuilder(BillingUtils.COST_FIELD).order(SortOrder.DESC)));
+    }
+
+    public BucketSortPipelineAggregationBuilder aggregateCostSortBucket(final int from, final int size) {
+        return aggregateCostSortBucket().from(from).size(size);
+    }
+
+    public SumBucketPipelineAggregationBuilder aggregateRunUsageSumBucket() {
+        return PipelineAggregatorBuilders.sumBucket(BillingUtils.RUN_USAGE_AGG,
+                aggsPath(BillingUtils.HISTOGRAM_AGGREGATION_NAME, BillingUtils.RUN_USAGE_AGG));
+    }
+
+    public SumBucketPipelineAggregationBuilder aggregateRunCostSumBucket() {
+        return PipelineAggregatorBuilders.sumBucket(BillingUtils.RUN_COST_AGG,
+                aggsPath(BillingUtils.HISTOGRAM_AGGREGATION_NAME, BillingUtils.RUN_COST_AGG, BillingUtils.COST_FIELD));
+    }
+
+    public SumBucketPipelineAggregationBuilder aggregateStorageCostSumBucket() {
+        return PipelineAggregatorBuilders.sumBucket(BillingUtils.STORAGE_COST_AGG,
+                aggsPath(BillingUtils.HISTOGRAM_AGGREGATION_NAME, BillingUtils.STORAGE_COST_AGG,
+                        BillingUtils.COST_FIELD));
+    }
+
+    public SumBucketPipelineAggregationBuilder aggregateCostSumBucket() {
+        return PipelineAggregatorBuilders.sumBucket(BillingUtils.COST_FIELD,
+                aggsPath(BillingUtils.HISTOGRAM_AGGREGATION_NAME, BillingUtils.COST_FIELD));
+    }
+
+    public AvgBucketPipelineAggregationBuilder aggregateStorageUsageAverageBucket() {
+        return PipelineAggregatorBuilders.avgBucket(BillingUtils.STORAGE_USAGE_AGG,
+                aggsPath(BillingUtils.HISTOGRAM_AGGREGATION_NAME, BillingUtils.STORAGE_USAGE_AGG));
+    }
+
+    private String aggsPath(final String... paths) {
+        return bucketsPath(BillingUtils.ES_DOC_AGGS_SEPARATOR, paths);
+    }
+
+    private String fieldsPath(final String... paths) {
+        return bucketsPath(BillingUtils.ES_DOC_FIELDS_SEPARATOR, paths);
+    }
+
+    private String bucketsPath(final String separator, final String[] paths) {
+        return String.join(separator, paths);
     }
 
     public Optional<Long> getRunUsageSum(final Aggregations aggregations) {
```

Comment:
Maybe we can change the return type to `Long` and use 0 as a default value?

As I can see right now at the calls sites of this and similar methods `.orElse(NumberUtils.LONG_ZERO)`  is used to retrieve the default value anyway

Reply:
Got rid of `Optional` from such methods signatures.